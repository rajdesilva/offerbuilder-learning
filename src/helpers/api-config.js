import { init as initApm } from "@elastic/apm-rum";

export const baseURL = {
  API_URL_BASE: process.env.REACT_APP_BASE_API_URL,
  LOGIN_URL: process.env.REACT_APP_LOGIN_URL,
  LOGOUT_URL: process.env.REACT_APP_LOGOUT_URL,
  PROFILE_URL: process.env.REACT_APP_USER_PROFILE_URL,
  HISTORY_URL: process.env.REACT_APP_USER_HISTORY_URL,
};

export const apm = initApm({
  // Set required service name (allowed characters: a-z, A-Z, 0-9, -, _, and space)
  serviceName: process.env.REACT_APP_APM_SERVICE_NAME,
  serverUrl: process.env.REACT_APP_APM_SERVICE_URL,

  // Set service version (required for sourcemap feature)
  serviceVersion: process.env.REACT_APP_APM_SERVICE_VERSION,
});