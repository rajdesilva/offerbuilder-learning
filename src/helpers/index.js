export * from './api-config';
export * from './history';
