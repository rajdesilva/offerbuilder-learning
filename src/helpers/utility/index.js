export { createSearchQueryURL } from './createSearchQueryURL';
export { isParentOrChildrenPropertyAdded } from './isParentOrChildrenPropertyAdded';
export { isPropertySelected } from './isPropertySelected';
export { mapBrands } from './mapBrands';
export { mapOnlySuppliers } from './mapOnlySuppliers';
export { parseFormDescriptionsAndImagesForOffer } from './parseFormDescriptionsAndImagesForOffer';
export { removePropertyForSelectedChannels } from './removePropertyForSelectedChannels';
export { removePropertyForSelectedSuppliers } from './removePropertyForSelectedSuppliers';
export { getParsedProperty } from './getParsedProperty';
export { getFormattedValue } from './getFormattedValue';
export { getValue } from './getValue';
export { parsePropertyMarketingInfoToFormData } from './parsePropertyMarketingInfoToFormData';
export { createOfferRequestObject } from './createOfferRequestObject';
export { getParsedOffers } from './getParsedOffers';
export { getListOfPropertiesForSupplier } from './getListOfPropertiesForSupplier';
export { getListOfPropertiesForChannels } from './getListOfPropertiesForChannels';
export { getInitializedState } from './getInitializedState';
export { getParsedNewOfferSettingsObject } from './getParsedNewOfferSettingsObject';
export { getListOfPropertiesForStorefront } from './getListOfPropertiesForStorefront';
export { checkIfMultipleChannelPresentForTarget } from './checkIfMultipleChannelPresentForTarget';
export { checkIfMultipleSupplierPresentForTarget } from './checkIfMultipleSupplierPresentForTarget';
export { checkIfMultipleStorefrontPresentForTarget } from './checkIfMultipleStorefrontPresentForTarget';
export { removePropertyForSelectedStorefront } from './removePropertyForSelectedStorefront';
export { checkIfMultipleBrandPresentForTarget } from './checkIfMultipleBrandPresentForTarget';
export { getListOfPropertiesForBrand } from './getListOfPropertiesForBrand';
export { getSearchStateFromRedux } from './getSearchStateFromRedux';
export { getObjectStringify } from './getObjectStringify';
export { parseOfferMarketingFromOffer } from './parseOfferMarketingFromOffer';
export { parseOfferMarketingLanguagesFromOffer } from './parseOfferMarketingLanguagesFromOffer';
export { getFilterData } from './getFilterData';
export { getSearchAndFilterQueryURL } from './getSearchAndFilterQueryURL';
export { getSearchQueryURLForEditAddProperties } from "./getSearchQueryURLForEditAddProperties";
export { isSavedOfferProperty } from './isSavedOfferProperty';
export { getParsedPropertiesForDeeplinkSettings } from "./getParsedPropertiesForDeeplinkSettings";
export { resetApplicationReduxAndStorage } from "./resetApplicationReduxAndStorage";
export { updateSettingsBrandWithSearchedValues } from "./updateSettingsBrandWithSearchedValues";
export { checkIfUserHasRole } from "./checkIfUserHasRole";
export { getUserOfferBuilderRole } from "./getUserOfferBuilderRole";
export { checkIfNewPropertiesAdded } from "./checkIfNewPropertiesAdded";
export { getAllBrandsWithoutKeysFromTempBrands } from "./getAllBrandsWithoutKeysFromTempBrands";
export { checkIfItemAlreadyPresentInList } from "./checkIfItemAlreadyPresentInList";
export { checkIfTravellingDateValidAsPerBookingDate } from "./checkIfTravellingDateValidAsPerBookingDate";
export { checkIfPropertiesHaveMultipleLocations } from "./checkIfPropertiesHaveMultipleLocations";
export { displayMargin } from "./displayMargin";
export { getFilterCount } from "./getFilterCount";
export { labelForBrands } from "./labelForBrands";
export { getDistanceFromLatLonInKm } from "./getDistanceFromLatLonInKm";
export { getTrustYouIdListFromProperties } from "./getTrustYouIdListFromProperties";
export { getSearchParamsToFetchMarketPrice } from "./getSearchParamsToFetchMarketPrice";
export { getMarketPriceForPropertyFromList } from "./getMarketPriceForPropertyFromList";
export { getImageFilterCount } from "./getImageFilterCount";
export { getSearchAndFilterImagesQueryUrl } from "./getSearchAndFilterImagesQueryUrl";
export { getImageName } from "./getImageName";
export { updateReduxWithOfferData } from "./updateReduxWithOfferData";
export { getOfferTypeFromPropertiesSelected } from "./getOfferTypeFromPropertiesSelected";