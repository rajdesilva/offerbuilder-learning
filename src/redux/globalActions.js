export const globalActions = {
  'SET_SUCCESS': 'SET_SUCCESS',
  'SET_ERROR': 'SET_ERROR',
  'SET_LANGUAGE': 'SET_LANGUAGE',
  'CLEAR': 'CLEAR',
}