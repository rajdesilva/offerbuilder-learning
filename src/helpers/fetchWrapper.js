import { message } from "antd";
import { createIntl, createIntlCache } from "@formatjs/intl";
import * as languages from "../langs/";
import { baseURL } from "./api-config";
import { history } from "./history";

// This is optional but highly recommended
// since it prevents memory leak
const cache = createIntlCache();

// Create the `intl` object
const intl = createIntl(
  {
    // Locale of the application
    // TODO:
    locale: "en",
    // Locale of the fallback defaultMessage
    defaultLocale: "en",
    messages: languages["en"],
  },
  cache
);
const defaultRequest = {
  credentials: "include",
  headers: {
    "Content-Type": "application/json",
  },
};
export const fetchWrapper = (url, payload) => {
  payload = {
    ...defaultRequest,
    ...payload,
  };
  return fetch(url, payload).then(
    (response) => {
      // Check that the response is valid and reject an error
      // response to prevent subsequent attempt to parse json
      if (!response.ok) {
        if (response.status === 500) {
          message.error(
            intl.formatMessage({
              id: "nemo.error500msg",
            })
          );
        } else if (response.status === 503) {
          message.error(
            intl.formatMessage({
              id: "nemo.error503msg",
            })
          );
        } else if (response.status === 504) {
          message.error(
            intl.formatMessage({
              id: "nemo.error504msg",
            })
          );
        } else {
          message.error(
            intl.formatMessage({
              id: "nemo.somethingWentWrongPleaseTryAgain",
            })
          );
        }
        return;
      }
      if (response.status === 403) {
        history.push("/access-denied");
      }
      return response.json ? response.json() : response;
    },
    (error) => {
      /**
       * user url, payload and error here for sending to the logger.
       */
      if (error instanceof TypeError) {
        window.location.href = baseURL.API_URL_BASE + "redirect";
      }
      console.error("fetch error : ", error);
    }
  );
};
