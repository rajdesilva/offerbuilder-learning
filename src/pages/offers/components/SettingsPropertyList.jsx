import React, { Fragment, useMemo } from "react";
import { Row, Table } from "antd";
import { useDispatch, useSelector } from "react-redux";
import { DeleteCartItem } from "../../browseSupply/components/DeleteCartItem";
import { displayMargin, getFormattedValue } from "../../../helpers/utility";
import { FormattedMessage } from "react-intl";
import { sortableElement, sortableContainer, sortableHandle } from 'react-sortable-hoc';
import { MenuOutlined } from '@ant-design/icons';
import arrayMove from 'array-move';
import { propertyCartActions } from "../../browseSupply/actions";
import styles from './css/SettingsPropertyList.module.less';

const DragHandle = sortableHandle(() => (
  <div className={styles.dragDiv}>
    <MenuOutlined className={styles.menuColor} />
  </div>
  
));

const SortableItem = sortableElement(props => <tr {...props} />);
const SortableContainer = sortableContainer(props => <tbody {...props} />);

export function SettingsPropertyList(props) {
  const dispatch = useDispatch();
  const selectedProperties = useSelector((state) =>
    window.getValue(state, "propertycart.cartItems")
  );
  const onSortEnd = ({ oldIndex, newIndex }) => {
    if (oldIndex !== newIndex) {
      const newData = arrayMove([].concat(selectedProperties), oldIndex, newIndex).filter(el => !!el);
      console.log('Sorted items: ', newData);
      dispatch({
        type: propertyCartActions.UPDATE_ENTIRE_CART_LIST,
        payload: newData,
      })
    }
  };

  const DraggableContainer = (props) => (
    <SortableContainer
      useDragHandle
      disableAutoscroll
      helperClass={styles["row-dragging"]}
      onSortEnd={onSortEnd}
      {...props}
    />
  );

  const DraggableBodyRow = ({ className, style, ...restProps }) => {
    // function findIndex base on Table rowKey props and should always be a right array index
    const index = selectedProperties ? selectedProperties.findIndex(x => x.key === restProps['data-row-key']) : 0;
    return <SortableItem index={index} {...restProps} />;
  };
  
  const columns = useMemo(
    () => [
      {
        title: '',
        width: "6%",
        className: styles['drag-visible'],
        render: () => <DragHandle />,
      },
      {
        title: <FormattedMessage id="nemo.properties" />,
        width: "25%",
        className: styles['drag-visible'],
        render: (text, row) => <div>{row.hotelName}</div>,
      },
      {
        title: <FormattedMessage id="nemo.location" />,
        className: "settings-property-list-item",
        width: "15%",
        render: (row) => (
          <Row>
            {row.city},&nbsp;
            {row.country} &nbsp;
          </Row>
        ),
      },
      {
        title: <FormattedMessage id="nemo.margin" />,
        dataIndex: "",
        className: styles["fs-fw-600"],
        width: "10%",
        render: (row) => (
          <Row>
            {displayMargin(row.lowestMargin, row.highestMargin) || (
              <FormattedMessage id="nemo.n/a" />
            )}
          </Row>
        ),
      },
      {
        title: <FormattedMessage id="nemo.supplier" />,
        className: "settings-property-list-item",
        width: "14%",
        render: (row) => (
          <Row>
            {row.supplier ? row.supplier.toUpperCase() : ""}-{row.channel}
          </Row>
        ),
      },

      {
        title: <FormattedMessage id="nemo.lcn" />,
        className: "settings-property-list-item",
        width: "8%",
        render: (row) => (
          <Row>{row.lcn ? <FormattedMessage id="nemo.yes" /> : ""}</Row>
        ),
      },
      {
        title: <FormattedMessage id="nemo.capitalPool" />,
        className: "settings-property-list-item",
        width: "17%",
        render: (row) => (
          <Fragment>
            {row.lcn ? (
              <Row>
                {row.remainingCapital ? (
                  getFormattedValue(row.remainingCapital)
                ) : (
                  <Fragment />
                )}{" "}
                <FormattedMessage id="nemo.eur" />
              </Row>
            ) : (
              <Fragment />
            )}
          </Fragment>
        ),
      },
      {
        width: "5%",
        render: (row) =>
          props.isForViewOnly ? (
            ""
          ) : (
            <Row justify="end">
              <DeleteCartItem row={row} isForSettings={true} />
            </Row>
          ),
      },
    ],
    [props.isForViewOnly]
  );

  return (
    <Table
      dataSource={selectedProperties}
      columns={columns}
      scroll={{ y: 340 }}
      className="settings-property-list"
      data-testid="property-list-table"
      pagination={false}
      rowKey="key"
      expandable={{
        expandIcon: () => <Fragment />,
      }}
      components={{
        body: {
          wrapper: DraggableContainer,
          row: DraggableBodyRow,
        },
      }}
    />
  );
}
