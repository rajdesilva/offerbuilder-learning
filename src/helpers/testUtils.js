import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render as rtlRender } from "@testing-library/react";
import { getValue } from "./utility";
import rootReducers from "../redux/rootReducers";
import * as languages from "../langs";
import { IntlProvider } from "react-intl";
const initialReducerState = {};
window.getValue = getValue;

window.google = {
  maps: {
    Marker: class {},
    Map: class {},
    LatLngBounds: class {},
    places: {
      Autocomplete: class {},
      AutocompleteService: class {},
      PlacesServiceStatus: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
      PlacesAutocomplete: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
    },

    MarkerClusterer: class {},
    Geocoder: class {},
  },
};
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

function render(
  ui,
  {
    initialState = initialReducerState,
    store = createStore(rootReducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <IntlProvider locale="en" messages={languages["en"]}>
          {children}
        </IntlProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
