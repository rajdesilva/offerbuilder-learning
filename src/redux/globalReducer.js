import { globalActions } from "./globalActions";
import produce from "immer";

const initialState = {
  error: "",
  success: "",
  language: "",
};
export const globalReducer = produce((draft, action) => {
  switch (action.type) {
    case globalActions.SET_ERROR:
      draft.success = "";
      draft.error = action.payload;
      return;
    case globalActions.SET_SUCCESS:
      draft.error = "";
      draft.success = action.payload;
      return;
    case globalActions.SET_LANGUAGE:
      draft.language = localStorage.getItem('language')
      return;
    case globalActions.CLEAR:
      return initialState;
    default:
      return draft;
  }
}, initialState);
