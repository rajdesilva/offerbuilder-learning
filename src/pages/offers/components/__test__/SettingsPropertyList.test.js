import React from "react";
import { createStore } from "redux";
import { render } from "../../../../helpers/testUtils";
import { MemoryRouter } from "react-router-dom";
import { SettingsPropertyList } from "../SettingsPropertyList";
import { tempStore } from "./tempStore";

describe("SettingsPropertyList component test", () => {
  test("create SettingsPropertyList components snapshot", () => {
    const store = createStore(() => ({ ...tempStore }));
    const wrapper = render(
      <MemoryRouter>
        <SettingsPropertyList />
      </MemoryRouter>,
      {
        store,
      }
    );
    const { container } = wrapper;
    expect(container.firstChild).toMatchSnapshot();
  });
});
