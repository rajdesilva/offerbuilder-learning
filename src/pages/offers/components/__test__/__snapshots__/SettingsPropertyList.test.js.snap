// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsPropertyList component test create SettingsPropertyList components snapshot 1`] = `
<div
  class="ant-table-wrapper settings-property-list"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table ant-table-fixed-header"
        data-testid="property-list-table"
      >
        <div
          class="ant-table-container"
        >
          <div
            class="ant-table-header"
            style="overflow: hidden;"
          >
            <table
              style="table-layout: fixed; visibility: hidden;"
            >
              <colgroup />
              <thead
                class="ant-table-thead"
              >
                <tr>
                  <th
                    class="ant-table-cell"
                  />
                  <th
                    class="ant-table-cell"
                  >
                    Properties
                  </th>
                  <th
                    class="ant-table-cell settings-property-list-item"
                  >
                    Location
                  </th>
                  <th
                    class="ant-table-cell"
                  >
                    Margin
                  </th>
                  <th
                    class="ant-table-cell settings-property-list-item"
                  >
                    Supplier
                  </th>
                  <th
                    class="ant-table-cell settings-property-list-item"
                  >
                    LCN
                  </th>
                  <th
                    class="ant-table-cell settings-property-list-item"
                  >
                    Capital Pool
                  </th>
                  <th
                    class="ant-table-cell"
                  />
                </tr>
              </thead>
            </table>
          </div>
          <div
            class="ant-table-body"
            style="overflow-y: scroll; max-height: 340px;"
          >
            <table
              style="table-layout: fixed;"
            >
              <colgroup>
                <col
                  style="width: 6%; min-width: 6%;"
                />
                <col
                  style="width: 25%; min-width: 25%;"
                />
                <col
                  style="width: 15%; min-width: 15%;"
                />
                <col
                  style="width: 10%; min-width: 10%;"
                />
                <col
                  style="width: 14%; min-width: 14%;"
                />
                <col
                  style="width: 8%; min-width: 8%;"
                />
                <col
                  style="width: 17%; min-width: 17%;"
                />
                <col
                  style="width: 5%; min-width: 5%;"
                />
              </colgroup>
              <tbody
                class="ant-table-tbody"
              >
                <tr
                  aria-hidden="true"
                  class="ant-table-measure-row"
                  style="height: 0px; font-size: 0px;"
                >
                  <td
                    style="padding: 0px; border: 0px; height: 0px;"
                  >
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                       
                    </div>
                  </td>
                  <td
                    style="padding: 0px; border: 0px; height: 0px;"
                  >
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                       
                    </div>
                  </td>
                  <td
                    style="padding: 0px; border: 0px; height: 0px;"
                  >
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                       
                    </div>
                  </td>
                  <td
                    style="padding: 0px; border: 0px; height: 0px;"
                  >
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                       
                    </div>
                  </td>
                  <td
                    style="padding: 0px; border: 0px; height: 0px;"
                  >
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                       
                    </div>
                  </td>
                  <td
                    style="padding: 0px; border: 0px; height: 0px;"
                  >
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                       
                    </div>
                  </td>
                  <td
                    style="padding: 0px; border: 0px; height: 0px;"
                  >
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                       
                    </div>
                  </td>
                  <td
                    style="padding: 0px; border: 0px; height: 0px;"
                  >
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                       
                    </div>
                  </td>
                </tr>
                <tr
                  data-row-key="DEMO_JEEVAN-0true"
                >
                  <td
                    class="ant-table-cell ant-table-cell-with-append"
                  >
                    <span
                      class="ant-table-row-indent indent-level-0"
                      style="padding-left: 0px;"
                    />
                    <div>
                      <span
                        aria-label="menu"
                        class="anticon anticon-menu"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="menu"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
                          />
                        </svg>
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div>
                      AJSG Hotel
                    </div>
                  </td>
                  <td
                    class="ant-table-cell settings-property-list-item"
                  >
                    <div
                      class="ant-row"
                    >
                      Vienna
                      , 
                      Austria
                        
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div
                      class="ant-row"
                    >
                      11 %
                    </div>
                  </td>
                  <td
                    class="ant-table-cell settings-property-list-item"
                  >
                    <div
                      class="ant-row"
                    >
                      NTP
                      -
                      NEMO_CH_06
                    </div>
                  </td>
                  <td
                    class="ant-table-cell settings-property-list-item"
                  >
                    <div
                      class="ant-row"
                    />
                  </td>
                  <td
                    class="ant-table-cell settings-property-list-item"
                  />
                  <td
                    class="ant-table-cell"
                  >
                    <div
                      class="ant-row ant-row-end"
                    >
                      <button
                        class="ant-btn ant-btn-text ant-btn-icon-only"
                        type="button"
                      >
                        <span
                          aria-label="delete"
                          class="anticon anticon-delete"
                          data-testid="delete-filled-icon"
                          role="img"
                          tabindex="-1"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="delete"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
